include/index.h

```diff
@@ -342,6 +363,17 @@ namespace diskann {
     bool _enable_tags = false;
     bool _normalize_vecs = false;  // Using normalied L2 for cosine.
 
+    // Filter Support
+
+    bool                            _filtered_index = false;
+    std::vector<std::vector<label>> _pts_to_labels;
```

Comment:
Can you please clarify this a bit further?

Reply:
I did some refactoring here to be more consistent ("label" is an attribute of a data point, a "filter" is a search that uses the dataset's labels). It gets ambiguous because broadly refer to this feature as "filtering", so sometimes we use the term "filter" to mean that "this code is in support of the 'filter' feature". I believe this will be an ongoing improvement on the codebase as the definitions become more clear over time.